tools/ci/sim_results.py

```diff
@@ -0,0 +1,152 @@
+        'scalarAttr': 'SELECT * FROM scalarAttr',
+        'vector': 'SELECT * FROM vector',
+        'vectorAttr': 'SELECT * FROM vectorAttr',
+        'vectorData': 'SELECT * FROM vectorData'
+    }
+
+    def __init__(self, vector_file: Path, scalar_file: Path):
+        self.__vector_file = vector_file
+        self.__scalar_file = scalar_file
+
+    def __lazy_load(self, attr: str, private_field: Optional[str] = None) -> pd.DataFrame:
+        if private_field is None:
+            private_field = f'__{attr}'
+
+        if getattr(self, private_field, None) is None:
+            if attr not in self.__sql_quieries:
+                raise KeyError(f'could not find query for attr: {attr}')
+            if attr not in self.__file_mapping:
+                raise KeyError(f'could not determine file type for attr: {attr}')
+            
+            match self.__file_mapping[attr]:
+                case self.File.SCALAR:
+                    file = self.__scalar_file
+                case self.File.VECTOR:
+                    file = self.__vector_file
+                case _:
+                    raise KeyError
+
+            with sqlite3.connect(file) as ctx:
+                df = pd.read_sql(self.__sql_quieries[attr], ctx)
+                setattr(self, private_field, df)
+
+        return getattr(self, private_field)
+    
+    @property
+    def run(self) -> pd.DataFrame:
+        return self.__lazy_load('run')
+
+    @property
+    def runAttr(self) -> pd.DataFrame:
+        return self.__lazy_load('runAttr')
+
+    @property
+    def runItervar(self) -> pd.DataFrame:
+        return self.__lazy_load('runItervar')
+
+    @property
+    def runParam(self) -> pd.DataFrame:
+        return self.__lazy_load('runParam')
+
+    @property
+    def scalar(self) -> pd.DataFrame:
+        return self.__lazy_load('scalar')
+
+    @property
+    def scalarAttr(self) -> pd.DataFrame:
+        return self.__lazy_load('scalarAttr')
+
+    @property
+    def vector(self) -> pd.DataFrame:
+        return self.__lazy_load('vector')
+
+    @property
+    def vectorAttr(self) -> pd.DataFrame:
+        return self.__lazy_load('vectorAttr')
+
+    @property
+    def vectorData(self) -> pd.DataFrame:
+        return self.__lazy_load('vectorData')
+    
+    @cached_property
+    def simtimeExp(self) -> int:
+        if self.run['simtimeExp'].shape[0] != 1:
+            raise RuntimeError('simtimeExp property only works for single runs')
+
+        return self.run['simtimeExp'][0]
+
+
+class SimResultsReader:
+    def __scan_results(self, scenerio_path: Path) -> Dict[str, Dict[str, Path]]:
+        if not isinstance(scenerio_path, Path):
+            raise TypeError
+        
+        scenerio_path = scenerio_path.resolve()
+        if not scenerio_path.is_dir():
+            raise FileNotFoundError(f'scenerio folder was not found: {scenerio_path}')
+        # sanity check
+        results_dir = scenerio_path.joinpath('results')
```

Comment:
For large simulation studies I often use custom result directories. Maybe the `results_dir` should become a parameter with *results* as default?